README.md

```diff
@@ -49,6 +49,100 @@ This will set up an Electrode webapplication which will have 2 of the above 6 mo
+#### 1. Create a new file `sw.js`,  
+```
+self.addEventListener("push", (event) => {
+  const title = "It worked!";
+  const options = {
+    body: "Great job sending that push notification!"
+  };
+  event.waitUntil(
+    self.registration.showNotification(title, options)
+  );
+});
+```
+[Sample file](https://github.com/electrode-io/electrode-boilerplate-universal-react-node/blob/master/client/sw.js)
+
+
+#### 2. Include this file in your webpack bundle by referencing it in `sw-config.js` :
+```
+module.exports = {
+  cache: {
+    importScripts: ['./sw.js']
+    }
+}
+```
+[Sample file](https://github.com/electrode-io/electrode-boilerplate-universal-react-node/blob/master/config/sw-config.js)
+
+
+#### 3. Now we can register this `push` event in the browser and install the new service worker:  
+```
+if ("serviceWorker" in navigator) {
+  navigator.serviceWorker.register("sw.js", { scope: "./" })
+  .then((registration) => {
+    // Service worker registration was successful
+  }
+}
+```
+[Sample file](https://github.com/electrode-io/electrode-boilerplate-universal-react-node/blob/master/client/register-service-worker.js)
+
+ The service worker is ready to accept `push` from the server. On receiving the push, it will provide the `notification` to the browser.
+
+## Instructions for sending a push notification  
+Now that we have our service worker up and running, we can send a `push` with the following steps:  
+
+#### 1. Requesting Permission and Subscribing Users  
+The code for requesting permission and subscribing users in done in your app's code, rather than the service worker code.
+```
+navigator.serviceWorker.ready.then((registration) => {
+  // Ask for user permission and subscribe
+  registration.pushManager.subscribe({ userVisibleOnly: true })
+    .then((subscription) => {
+      // Successfully subscribed
+      this.setState({
+        subscription,
+        subscribed: true
+      });
+    });
+});
+```
+[Sample subscription](https://github.com/electrode-io/electrode-boilerplate-universal-react-node/blob/master/client/components/push-notifications.jsx#L73-L86)  
+
+Typically, after the user subscribes, we send the subscription information to the server and the server uses the `subscriptionId` to trigger a notification.
+
+#### 2. Sending Messages  
+Sending message is as easy as executing a curl command. The curl command contains the `subscriptionId`, `API_KEY` (Your cloud messaging API key from [firebase](https://console.firebase.google.com)) and [GCM_ENDPOINT](https://github.com/electrode-io/electrode-boilerplate-universal-react-node/blob/master/client/components/push-notifications.jsx#L145-L146)
+
+Alternatively, you can also send notifications from the service worker:
+```
+sendNotification() {
+   const title = 'This Notification'
+   const body = 'Is brought to you by your server worker!'
+   const options = {body};
+   navigator.serviceWorker.ready.then((registration) => {
```

Comment:
Might be worth explaining that `navigator.serviceWorker.ready` is a Promise that will resolve once a service worker is registered, and it returns a reference to the active [ServiceWorkerRegistration](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration)